src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,129 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.model.DataProcessingException;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+
+    @Override
+    public Book create(Book book) {
+        Connection connection = ConnectionUtil.getConnection();
+        String sqlInsert = "INSERT INTO books(title, price) VALUES (?, ?)";
+        try {
+            PreparedStatement statement = connection.prepareStatement(
+                    sqlInsert,
+                    Statement.RETURN_GENERATED_KEYS
+            );
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            int affectedRows = statement.executeUpdate();
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+            if (affectedRows < 1) {
+                throw new RuntimeException("Book wasn't added. 0 rows were affected");
+            }
+            statement.close();
+            connection.close();
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error while adding book", e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        Connection connection = ConnectionUtil.getConnection();
```

Comment:
you should use `try-with-resources` for both `Connection` and `PreparedStatement`